src/components/AttendantForm.test.jsx

```diff
@@ -0,0 +1,151 @@
+import AttendantForm from "./AttendantForm";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import ErrorMessages from "../constants/errorMessages";
+
+const jobTitles = ["Accountant", "Engineer", "Lawyer", "Teacher"];
+
+const fillAndSubmitForm = async () => {
+  userEvent.type(screen.getByTestId("name"), "John");
+  userEvent.type(screen.getByTestId("last-name"), "Doe");
+  userEvent.selectOptions(await screen.findByTestId("job-title"), "Engineer");
+  userEvent.type(screen.getByTestId("age"), "40");
+  userEvent.click(screen.getByTestId("submit"));
+};
+
+describe("<AttendantForm />", () => {
+  let props;
+
+  beforeEach(() => {
+    props = {
+      isAttendantsLoading: false,
+      isJobTitlesLoading: false,
+      submitAttendant: jest.fn(),
+      jobTitles: jobTitles,
+      jobTitlesApiError: null,
+    };
+  });
+
+  it("renders correctly when data is loaded", async () => {
+    const { container } = render(<AttendantForm {...props} />);
+
+    expect(screen.queryByTestId("job-titles-loader")).not.toBeInTheDocument();
+    expect(container).toMatchSnapshot();
+  });
+
+  it("renders job title loader", async () => {
+    render(
+      <AttendantForm
+        {...props}
+        isJobTitlesLoading={true}
+      />
+    );
+
+    expect(screen.getByTestId("job-titles-loader")).toBeInTheDocument();
+  });
+
+  it("renders error message when get job titles API call fails", async () => {
+    render(
+      <AttendantForm
+        {...props}
+        jobTitlesApiError={[ErrorMessages.FAILED_TO_GET_JOB_TITLES]}
+      />
+    );
+
+    expect(
+      screen.getByText(ErrorMessages.FAILED_TO_GET_JOB_TITLES)
+    ).toBeInTheDocument();
+  });
+
+  it("does not render select job title dropdown when get job titles API call fails", () => {
+    render(
+      <AttendantForm
+        {...props}
+        jobTitlesApiError={[ErrorMessages.FAILED_TO_GET_JOB_TITLES]}
+      />
+    );
+
+    expect(screen.queryByTestId("job-title")).not.toBeInTheDocument();
```

Comment:
About test ids - react testing library suggests writing tests that reflect how the user would use the app which implies using querying by roles, labels, text (everything except test ids).

Test ids are fine for some use cases (there are even some performance benefits to doing so especially good for marketplace monolith) and for the sake of this exercise I would suggest trying the former (non-test id) method, to get the hang of it.

Reply:
Good point! Tried to use other selectors as labels, texts, placehoderText.
I just can not figure out how to select `job-title` dropdown option without using test-id. Tried by text and by role but no luck 🤔